src/coreclr/pal/src/misc/sysinfo.cpp

```diff
@@ -575,12 +575,16 @@ PAL_GetLogicalProcessorCacheSizeFromOS()
         // this method to determine cache size.
         //
         size_t level;
+        char path_to_size_file[] =  "/sys/devices/system/cpu/cpu0/cache/index-/size";
+        char path_to_level_file[] =  "/sys/devices/system/cpu/cpu0/cache/index-/level";
+        int index = 40;
```

Comment:
A nit - maybe worth adding an assert that the index corresponds to the `-` character in the path.